nntrainer/layers/layer.cpp

```diff
@@ -84,6 +100,74 @@ void Layer::copy(std::shared_ptr<Layer> l) {
   this->num_outputs = l->num_outputs;
 }
 
+sharedConstTensors Layer::forwarding_with_val(sharedConstTensors input,
+                                   sharedConstTensors in) {
+
+  for(unsigned int i=0;i<num_inputs;++i){
+    net_input[i]->var = *input[i];
+  }
+
+  if (num_outputs == 0)
```

Comment:
is this possible? (I assume this function can only run once layer is initialized).

Reply:
I was thinking set everything based 1 is not good at that time. Currently we just set as 1 with no other questions. I guess it is not good. but we still doing with 1 so, it is not possible. I will delete it. done.